internal/auth/key_loader.go

```diff
@@ -0,0 +1,232 @@
+	}
+
+	return
+}
+
+// recoverFromLoadApiKeyError tries to recover from an error during key loading. We treat different errors differently.
+// For example if the path is the default path and the error is os.ErrNotExist, this could be just the first start of Clouditor.
+// So we only treat this as an information that we will create a new key, which we will save, based on the config.
+//
+// If the user specifies a custom path and this one does not exist, we will report an error
+// here.
+func (l *keyLoader) recoverFromLoadApiKeyError(err error, defaultPath bool) (key *ecdsa.PrivateKey) {
+	// In any case, create a new temporary API key
+	key, _ = ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+
+	if defaultPath && errors.Is(err, os.ErrNotExist) {
+		log.Infof("API key does not exist at the default location yet. We will create a new one")
+
+		if l.saveOnCreate {
+			// Also save the key in this case, so we can load it next time
+			err = saveKeyToFile(key, l.path, l.password)
+
+			// Error while error handling, meh
+			if err != nil {
+				log.Errorf("Error while saving the new API key: %v", err)
+			}
+		}
+	} else if err != nil {
+		log.Errorf("Could not load key from file, continuing with a temporary key: %v", err)
+	}
+
+	return key
+}
+
+// loadKeyFromFile loads an ecdsa.PrivateKey from a path. The key must in PEM format and protected by
+// a password using PKCS#8 with PBES2.
+func loadKeyFromFile(path string, password []byte) (key *ecdsa.PrivateKey, err error) {
+	var (
+		keyFile string
+	)
+
+	keyFile, err = expandPath(path)
+	if err != nil {
+		return nil, fmt.Errorf("error while expanding path: %w", err)
+	}
+
+	if _, err = os.Stat(keyFile); os.IsNotExist(err) {
+		return nil, fmt.Errorf("file does not exist (yet): %w", err)
+	}
+
+	// Check, if we already have a persisted API key
+	f, err := os.OpenFile(keyFile, os.O_RDONLY, 0600)
+	if err != nil {
+		return nil, fmt.Errorf("error while opening the file: %w", err)
+	}
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			log.Errorf("Error while closing file: %v", err)
+		}
+	}()
+
+	data, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("error while reading file content: %w", err)
+	}
+
+	key, err = ParseECPrivateKeyFromPEMWithPassword(data, password)
+	if err != nil {
+		return nil, fmt.Errorf("error while parsing private key: %w", err)
+	}
+
+	return key, nil
+}
+
+// saveKeyToFile saves an ecdsa.PrivateKey to a path. The key will be saved in PEM format and protected by
+// a password using PKCS#8 with PBES2.
+func saveKeyToFile(apiKey *ecdsa.PrivateKey, keyPath string, password string) (err error) {
+	keyPath, err = expandPath(keyPath)
+	if err != nil {
+		return fmt.Errorf("error while expanding path: %w", err)
+	}
+
+	// Check, if we already have a persisted API key
+	f, err := os.OpenFile(keyPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0600)
+	if err != nil {
+		return fmt.Errorf("error while opening the file: %w", err)
+	}
+	defer f.Close()
```

Comment:
For me this is still a philosophical question, why the error of f.Close in a deferred call should be handled at all. There are two circumstances in which an error is thrown from `Close`:
- The file is nil, i.e. was never opened -> thus not need to be closed
- The file is already closed -> does not need to be closed again

So I think there is no real need to log or otherwise handle this error.